homeassistant/components/template/light.py

```diff
@@ -343,3 +385,22 @@ def update_state(self):
             except TemplateError as ex:
                 _LOGGER.error(ex)
                 self._state = None
+
+    @callback
+    def update_temperature(self):
+        """Update the temperature from the template."""
+        if self._temperature_template is not None:
```

Comment:
Use a guard clause: `if self._temperature_template is None: return`. Even better would be to not even call this function if not necessary.